### ISSUE TITLE
[Android] Missing support for Jetpack Glance

### ISSUE THREAD

**Issue by paul678** [ABausG/home_widget#134]
Hey, first of all many thanks for an awesome plugin 🙏 I personally started using it to include widget support for my app [Habit It](https://habitit.thinkoutside.app/) and it looks great!

However I've seen it's missing support for [Jetpack Glance](https://developer.android.com/jetpack/androidx/releases/glance), Google's [new way](https://android-developers.googleblog.com/2021/12/announcing-jetpack-glance-alpha-for-app.html) to describe home widgets.

From all I could gather it seems that the view create by Glance will refresh only under certain conditions and the old event broadcast is [no longer recommended](https://stackoverflow.com/questions/70502137/how-to-update-the-appwidgetstate-of-a-compose-glance-widget-from-a-configuration#comment124919114_70659416) by Google 🤔 

I've went ahead and did an implementation [here](https://github.com/paul678/home_widget/tree/glance_support) that uses the recommended approach but it's lacking an example and some proper documentation (also I'm still testing the feature for a couple of days so I might push other changes)

**Comment by ABausG** [comment]
Great. Is there a way of offering both solutions side by side? I guess the underlaying principle of using shared preferences would be the same and it would be mainly moving (documentation) over from `HomeWidgetProvider`?

Looking forward to a PR

**Comment by paul678** [comment]
Sorry for ghosting this issue but I was, and still am really busy and could not work on any of my personal projects for some time now. 
I will resume somewhere in September and come back to this issue as well

**Comment by Musta-Pollo** [comment]
That would be great ❤️ !

**Comment by myxzlpltk** [comment]
Jetpack glance implementation will not affect ios right?

**Comment by paul678** [comment]
Should not affect iOS and neither the external Flutter API I would say

**Comment by tenhobi** [comment]
Hey, any updates on this? 🙏

**Comment by armandsLa** [comment]
I managed to solve the widget update issue. It turns out that starting with version `1.0.0-beta01`, the `provideGlance` method won't be called after every update. This method is only called for the initial load, and all subsequent updates will only reload the state. This post explains well how it works and also shows how to implement a custom `GlanceStateDefinition`. https://stackoverflow.com/questions/77088363/android-jetpack-glance-1-0-0-problems-updating-widget

**Comment by lora-explora** [comment]
> I managed to solve the widget update issue. It turns out that starting with version `1.0.0-beta01`, the `provideGlance` method won't be called after every update. This method is only called for the initial load, and all subsequent updates will only reload the state. This post explains well how it works and also shows how to implement a custom `GlanceStateDefinition`. https://stackoverflow.com/questions/77088363/android-jetpack-glance-1-0-0-problems-updating-widget
> 
> Unfortunately, the update README is incorrect. As explained above, it's not possible to update `___FAKE_UPDATE___` preference to force widget reload and then call `HomeWidget.getWidgetData<String>('id', defaultValue: data)` from `provideGlance` method to retrieve the data. `onUpdate` method will be called, but `provideGlance` wont. All of the data must be passed to the new state.

Cool that you're looking into this! Could you please give another pointer how the data is passed to the new state? Do you still call 'updateAppWidgetState'? 
I tried the stack overflow response and I currently call this in onUpdate() ```
updateAppWidgetState(context, stateDefinition, glanceId) { data ->
                        data
                    }
```
I can see that my glanceState is getting updated with the correct data but provideGlance is still not called.. And I can't figure out why.

**Comment by ABausG** [comment]
@armandsLa Thanks for digging into this. Would you be able to assist in improving this? Just glancing over the stackoverflow article I guess in an ideal world HomeWidget would provide a `HomeWidgetGlanceState` that could be used?

**Comment by ABausG** [comment]
@armandsLa  thank you for finding that stack overflow link. I just finished implementing some more Glance Logic so updating/accessing data will be super nice to do with Glance!!

**Comment by ABausG** [comment]
This is now released in 0.5.0
